server/models/Availability.js

```diff
@@ -0,0 +1,97 @@
+const mongoose = require("mongoose");
+
+const availabilitySchema = new mongoose.Schema({
+  profileId: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: 'Profile'
+  },
+  schedule: {
```

Comment:
A bit opinionated: I don't think there is a reason nest these fields under `schedule` - imo usage-wise `availability.monday` seems more appealing than `availability.schedule.monday`.